routers/user.py

```diff
@@ -1,39 +1,40 @@
-from fastapi import status,APIRouter
+from fastapi import status, APIRouter
 from fastapi.responses import JSONResponse
 from utils.jwt_manager import create_token
-from schemas.user import User,UserBase,UserCreate
+from schemas.user import User, UserBase, UserCreate
 from config.database import Session
 from services.user import UserService
 from services.auth import Auth
 
 user_router = APIRouter()
+db = Session()
+
+
+@user_router.post('/users', tags=['Auth'], response_model=User, status_code=status.HTTP_200_OK)
+def create_user(user: UserCreate):
+    check_user_exists(user)
 
-@user_router.post('/users',tags=['Auth'],response_model=User,status_code=status.HTTP_200_OK)
-def create_user(user:UserCreate):
-    
-    db = Session()
-    
-    result = UserService(db).get_user_by_email(email=user.email)
-    
-    if result:
-    
-        return JSONResponse(status_code=status.HTTP_400_BAD_REQUEST,content={"message":"User already exists"})
-    
     UserService(db).create_user(user)
-    
-    return JSONResponse(status_code=status.HTTP_200_OK,content={"message":"User created"})
-    
-
-@user_router.post('/login',tags=['Auth'],status_code=status.HTTP_200_OK)
-def login(user:UserCreate):
-    
-    db = Session()
-    result = UserService(db).get_user_by_email(email=user.email)
-    
-    if not (result and Auth().verify_password(user.password,result.password)):
-        
-        return JSONResponse(status_code=status.HTTP_401_UNAUTHORIZED,content={"message":"Unauthorized"})
-    
-    token:str = create_token(user.dict())
-    
-    return JSONResponse(status_code=status.HTTP_200_OK,content=token)
\ No newline at end of file
+
+    return JSONResponse(status_code=status.HTTP_200_OK, content={"message": "User created"})
+
+
+def check_user_exists(user):
+    if UserService(db).get_user_by_email(email=user.email):
+        return JSONResponse(status_code=status.HTTP_400_BAD_REQUEST, content={"message": "User already exists"})
+
+
+@user_router.post('/login', tags=['Auth'], status_code=status.HTTP_200_OK)
+def login(user: UserCreate):
+    validate_password(user)
+
+    token: str = create_token(user.dict())
```

Comment:
duda 🤔 , cual es el tipo de dato exacto que debe ser el token?

ya que aqui indicas que `token` debe ser de tipo `string` pero el método `create_token` retorna un `dic`.

---

y  el IDE me marco un error, busque y me indica el `dict` esta **deprecated** para la versión 2 de `pydantic`

https://docs.pydantic.dev/latest/migration/#changes-to-pydanticbasemodel

pero desconozco si se hizo así por alguna razón, aqui es a tu consideración bro